### ISSUE TITLE
fb.me/react-async-component-lifecycle-hooks should probably link somewhere useful

### ISSUE THREAD

**Issue by EdwardDrapkin** [reactjs/reactjs.org#658]
Right now, there are warning raised for soon-to-be-removed lifecycle methods that instruct users (like myself) to head to this URL for more information: fb.me/react-async-component-lifecycle-hooks

That URL currently resolves to a fairly massive ticket that's been closed for months, has a lot of developmental discussions, doesn't clearly lay out what the new API is, and doesn't offer any migration path.  This is not only unhelpful, but I feel like it will lead people to loading the URL, bailing, and ignoring the warnings, because that's exactly what I want to do right now.  It's not even clear if the RFC linked in the ticket is the format/API that got accepted! 

As a side note, is there even any documentation about the reasoning for the lifecycle changes and a documentated migration path that doesn't require I invest several hours deciphering issue comments and source code?   If there isn't why is this warning being raised already?

Sorry if this is the wrong repo, but it seems like the right place because this is the documentation repo, right?

**Comment by mackenziemance** [comment]
@EdwardDrapkin https://github.com/reactjs/rfcs/blob/master/text/0006-static-lifecycle-methods.md

**Comment by gaearon** [comment]
Alphas aren’t meant to be used by anyone except people who are willing to deal with incomplete or missing documentation. That URL will be updated to point to a proper location in a stable release.

You shouldn’t be using alphas for anything other than pure experimentation. There are known bugs, unstable APIs, and missing links in them.

**Comment by bvaughn** [comment]
I agree with Dan that this isn't really a valid issue, but I am curious @EdwardDrapkin- can you clarify how you are seeing this warning?

Even our [16.3 alpha](https://unpkg.com/react-dom@16.3.0-alpha.1/cjs/react-dom.development.js) has the `warnAboutDeprecatedLifecycles` flag turned off, so this warning should not have been shown in the console:

```js
// Warn about deprecated, async-unsafe lifecycles; relates to RFC #6:
var warnAboutDeprecatedLifecycles = false;
```

We don't intend to enable these deprecation warnings until the following 16.4 release.

**Comment by alexkrolick** [comment]
It's in the 16.4 alpha that you can get from unpkg or npm: https://unpkg.com/react@16.4.0-alpha.0911da3/umd/

**Comment by bvaughn** [comment]
Ah. That's not even an alpha release. (It's not tagged "next" in NPM.)

The flag is also disabled in that build you linked to [here](https://unpkg.com/react-dom@16.4.0-alpha.0911da3/cjs/react-dom.development.js) and [here](https://unpkg.com/react-dom@16.4.0-alpha.0911da3/umd/react-dom.development.js). (Neither the flag nor the warning message appear in the `react` package you linked to; they're in `react-dom`.)

Am I missing something?

**Comment by alexkrolick** [comment]
https://codesandbox.io/s/q8pmnwrrp4

![image](https://user-images.githubusercontent.com/1571667/37076365-2423bda0-218b-11e8-8627-c5fbb9f19063.png)

**Comment by bvaughn** [comment]
That's not the warning mentioned in this PR (which references  fb.me/react-async-component-lifecycle-hooks). That's the strict mode warning, which exists in 16.3 too (if you're using `React.StrictMode` or `React.unstable_AsyncMode`).

**Comment by EdwardDrapkin** [comment]
I installed the version listed as a peer dependency in React Native 0.54 as part of evaluating that upgrade.

https://github.com/facebook/react-native/blob/0.54-stable/package.json#L146

**Comment by gaearon** [comment]
Let’s reopen this. If RN already enables this warning it sounds bad.

**Comment by bvaughn** [comment]
Ah! Thank you for clarifying, @EdwardDrapkin!

I've reached out to @grabbou about this and we've fixed the warning for the React Native 0.54 branch. I didn't intend for it to be enabled there before `react` and `react-dom` 16.3 release was out. My apologies.

I'm going to close this issue now as the warning has been disabled [here](https://github.com/facebook/react-native/blob/0.54-stable/Libraries/Renderer/shims/ReactFeatureFlags.js#L15) and Mike will be releasing an update momentarily.
